inference-engine/src/transformations/src/transformations/common_optimizations/interpolate_sequence_fusion.cpp

```diff
@@ -0,0 +1,193 @@
+        for (const auto& consumer : output.get_target_inputs()) {
+            if (consumer.get_node() != snd.get()) return false;
+        }
+    }
+
+    return compatible_attrs(fst->get_attrs(), snd->get_attrs()) && is_candidate_for_fusion(fst) && is_candidate_for_fusion(snd) &&
+           compatible_axes(get_interpolated_axes(fst), get_interpolated_axes(snd));
+}
+
+ngraph::NodeVector subgraph_for_sizes_calculation_mode(const std::shared_ptr<opset8::Interpolate>& fst, const std::shared_ptr<opset8::Interpolate>& snd) {
+    const auto fst_axes = get_interpolated_axes(fst);
+    const auto snd_axes = get_interpolated_axes(snd);
+    const auto fst_sizes = std::dynamic_pointer_cast<opset8::Constant>(fst->input_value(1).get_node_shared_ptr())->cast_vector<int64_t>();
+    const auto snd_sizes = std::dynamic_pointer_cast<opset8::Constant>(snd->input_value(1).get_node_shared_ptr())->cast_vector<int64_t>();
+
+    std::vector<std::pair<int64_t, int64_t>> axes_and_sizes;
+    for (size_t i = 0; i < fst_axes.size(); ++i) {
+        axes_and_sizes.emplace_back(std::make_pair(fst_axes[i], fst_sizes[i]));
+    }
+    for (size_t i = 0; i < snd_axes.size(); ++i) {
+        axes_and_sizes.emplace_back(std::make_pair(snd_axes[i], snd_sizes[i]));
+    }
+    std::sort(axes_and_sizes.begin(),
+              axes_and_sizes.end(),
+              [](const std::pair<int64_t, int64_t>& a, const std::pair<int64_t, int64_t>& b) {
+                  return a.first < b.first;
+              });
+    std::vector<int64_t> new_axes;
+    std::vector<int64_t> new_sizes;
+    for (const auto& as : axes_and_sizes) {
+        new_axes.emplace_back(as.first);
+        new_sizes.emplace_back(as.second);
+    }
+
+    auto new_sizes_node = opset8::Constant::create(element::i64, {new_sizes.size()}, new_sizes);
+    auto new_axes_node = opset8::Constant::create(element::i64, {new_axes.size()}, new_axes);
+    auto new_sizes_cast = std::make_shared<opset8::Convert>(new_sizes_node, element::f32);
+    auto shape_node = std::make_shared<opset8::ShapeOf>(fst->input_value(0));
+
+    auto gather_axis_node = opset8::Constant::create(element::i64, {1}, std::vector<int64_t>{0});
+    auto gather_node = std::make_shared<opset8::Gather>(shape_node, new_axes_node, gather_axis_node);
+    auto cast_shape_to_float = std::make_shared<opset8::Convert>(gather_node, element::f32);
+
+    auto div_node = std::make_shared<opset8::Divide>(new_sizes_cast, cast_shape_to_float);
+
+    auto new_interpolate = fst->clone_with_new_inputs({fst->input_value(0), new_sizes_node, div_node, new_axes_node});
+
+    return {new_sizes_node, new_axes_node, new_sizes_cast, shape_node, gather_axis_node, gather_node, cast_shape_to_float, div_node, new_interpolate};
+}
+
+ngraph::NodeVector subgraph_for_scales_calculation_mode(const std::shared_ptr<opset8::Interpolate>& fst, const std::shared_ptr<opset8::Interpolate>& snd) {
+    const auto fst_axes = get_interpolated_axes(fst);
```

Comment:
We don't check rank here

Reply:
We needn't it, because at this time we have a **static** input rank. And we have a **static** input rank here, because this function will be called only if `can_be_fused(fst_interpolate, snd_interpolate)` returns `true`.